pkg/manager/impl/execution_manager.go

```diff
@@ -234,26 +234,36 @@ func createTaskDefaultLimits(ctx context.Context, task *core.CompiledTask,
 	}
 	taskResourceLimits := runtimeInterfaces.TaskResourceSet{}
 
-	// For resource values, we prefer to use the limits set in the application config over the set resource values.
-	if !configResourceLimits.CPU.IsZero() {
+	// For resource values, we prefer to set unset limits to the request values over the config values.
+	if cpuIndex >= 0 {
+		cpuLimit = resourceRequestEntries[cpuIndex].Value
```

Comment:
why do we parse this multiple times? Why not parse it once at load time and then use? this will cause failuers much later, incase of a bug in the config?